### PR TITLE
Fixes problems with weighted cross-entropy

### PR DESCRIPTION
**Pull request by franchuterivera**
address #248

### CODE REVIEW

**Review by ravinkohli — APPROVED** [review]
Hey, this PR looks great. I think this can be merged. Thanks for the added tests to ensure we don't mess this up going forward.

File: autoPyTorch/utils/implementations.py [hunk 1]
```diff
@@ -50,10 +50,8 @@ class LossWeightStrategyWeightedBinary():
     def __call__(self, y: Union[np.ndarray, torch.Tensor]) -> np.ndarray:
         if isinstance(y, torch.Tensor):
             y = y.detach().cpu().numpy() if y.is_cuda else y.numpy()
-        if isinstance(y[0], str):
-            y = y.astype('float64')
         counts_one = np.sum(y, axis=0)
-        counts_zero = counts_one + (-y.shape[0])
+        counts_zero = y.shape[0] - counts_one
         weights = counts_zero / np.maximum(counts_one, 1)
```
Comment (nabenabe0928) [inline autoPyTorch/utils/implementations.py:55]: Where do we use those weights?
If we weight the data with label 0 by this weights, `counts_zero / y.shape[0]` seems correct to me.

**Review by nabenabe0928 — COMMENTED** [review]
Thanks for the PR.
I put a comment regarding the validity of the calculation.
Please check the current implementation.

File: autoPyTorch/utils/implementations.py (hunk 1 above)
Comment (franchuterivera) [reply autoPyTorch/utils/implementations.py:55]: This follows the recommendation from pytorch, please look into [this](https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html).

The problem nevertheless here was that we had negative weights. For this reason I changed line 54. Line 55 Looks ok for me, but let me know if you find otherwise.

Comment (franchuterivera) [reply autoPyTorch/utils/implementations.py:56]: Thanks for the suggestion, I have done so now.
